### PR TITLE
Use remote file dialog for database and server properties dialogs

### PR DESCRIPTION
**Pull request by corivera**
Currently we only use the local file system dialog for some of the database and server properties path & file fields, but many SQL instances are on other machines. This change adds the same remote file dialog used for Attach and Restore to the properties dialogs.

As part of this work I had to add a flag to the file browser to be able to only show folders, since some of the properties fields are only folders, like the default database file location for example. Anytime the file browser is opened in folder-only mode the file filters box gets hidden and the other controls expand to fill the space.

With file filter:
DialogWithFilter

Without file filter:
DialogWithoutFilter

I also added a small fix for handling undefined values in the Attach Database dialog that I was hitting during some testing.

Currently waiting on the related changes in STS and the dataprotocol-client to be checked in before merging this.
https://github.com/microsoft/sqlops-dataprotocolclient/pull/95
https://github.com/microsoft/sqltoolsservice/pull/2238

### CODE REVIEW

**Comment by corivera** [comment]
Related issues:
https://github.com/microsoft/azuredatastudio/issues/24318
https://github.com/microsoft/azuredatastudio/issues/24313

I'll leave those open until the release branch ports are taken care of.

File: src/sql/azdata.d.ts [hunk 1]
```diff
@@ -2273,7 +2273,7 @@ declare module 'azdata' {
 	// File browser interfaces  -----------------------------------------------------------------------
 
 	export interface FileBrowserProvider extends DataProvider {
-		openFileBrowser(ownerUri: string, expandPath: string, fileFilters: string[], changeFilter: boolean): Thenable<boolean>;
+		openFileBrowser(ownerUri: string, expandPath: string, fileFilters: string[], changeFilter: boolean, showFoldersOnly?: boolean): Thenable<boolean>;
```
Comment (Charles-Gagnon) [inline src/sql/azdata.d.ts:2276]: This can go into proposed too

Comment (Charles-Gagnon) [reply src/sql/azdata.d.ts:2276]: e.g. it'll look like this in the proposed file

```
	export interface FileBrowserProvider extends DataProvider {
		openFileBrowser(ownerUri: string, expandPath: string, fileFilters: string[], changeFilter: boolean, showFoldersOnly?: boolean): Thenable<boolean>;
	}
```

(then add the doc comments to both versions)

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -2068,9 +2068,14 @@ declare module 'azdata' {
...
 		 * @param fileFilters The filters used to limit which files are displayed in the file browser
+		 * @param showFoldersOnly Optional argument to specify whether the browser should only show folders
 		 * @returns The path of the file chosen from the dialog, and undefined if the dialog is closed without selecting anything.
 		 */
-		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[]): Thenable<string | undefined>;
+		export function openServerFileBrowserDialog(connectionUri: string, targetPath: string, fileFilters: FileFilters[], showFoldersOnly?: boolean): Thenable<string | undefined>;
+	}
+
+	export interface FileBrowserProvider extends DataProvider {
+		openFileBrowser(ownerUri: string, expandPath: string, fileFilters: string[], changeFilter: boolean, showFoldersOnly?: boolean): Thenable<boolean>;
```
Comment (Charles-Gagnon) [inline src/sql/azdata.proposed.d.ts:2078]: If you could add a doc comment for this (and then copy it over minus the new param to azdata.d.ts) that would be fantastic 😄

**Review by Charles-Gagnon — APPROVED** [review]
Looks good once the STS & protocol changes are brought in

**Review by brian-harris — APPROVED** [review]

**Pull request merged**